src/lib.rs

```diff
@@ -239,13 +240,28 @@ where
     where
         F: FnMut(&T) -> R + 'static + Sync + Send,
         R: PartialEq + 'static + Sync + Send,
+    {
+        self.try_map(move |v| Ok(map(v))).ok().unwrap()
+    }
+
+    /// Creates a new refreshable from this one by applying a fallible mapping function to the value.
+    ///
+    /// This can be used to narrow the scope of the refreshable value. Updates to the initial refreshable value will
+    /// propagate to the mapped refreshable value, but the mapped refreshable's subscriptions will only be invoked if
+    /// the mapped value actually changed.
+    pub fn try_map<F, R>(&self, mut map: F) -> Result<Refreshable<R, E>, E>
```

Comment:
Same comment here as with subscription, but it is probably worth explicitly calling out what happens if subsequent invocations return a failure (it looks like we just maintain the old value).

Also, I'm kind of curious when we'd want something like this vs just returning a `Refreshable<Result<...>>`? If the mapping failed it seems like that's something we'd want to propagate downstream rather than just continue to provide the old value?

Reply:
We actually want to propagate upstream - in the standard case the "root" refreshable is a witchcraft-server service's runtime config, and errors are reported in a health check since they're generally caused by bad config.